### ISSUE TITLE
Display warning setApprovalForAll is called on an NFT collection

### ISSUE THREAD

**Issue by onyb** [brave/brave-browser#24268]
|Twitter | https://twitter.com/mulligan/status/1548383796126289921 |
-|-

`setApprovalForAll` is becoming a common exploit to drain all NFTs of an owner in a particular collection. We should display an appropriate warning when such an approval is requested.

![](https://user-images.githubusercontent.com/3684187/181021149-a51365ab-49c8-4ca3-9ad4-cc668b37531c.png)

### Tech spec

Use the `ABIDecoder` to parse the above transaction into a dedicated `TransactionType`. On the frontend, use the same warning style that we use for unlimited ERC20 approvals.

**Comment by diracdeltas** [comment]
@onyb are you opening these issues for iOS as well?

**Comment by onyb** [comment]
@diracdeltas No. I'd like to do the integration on desktop first and later create follow-up issues for ios/android teams. Just kept the `OS/Desktop` label for now.

**Comment by Douglashdaniel** [comment]
Accidentally closed this out with the wrong PR, Re-openeing!
